src/paths.jl

```diff
@@ -13,13 +13,54 @@ end
 
 #Polygon methods
 
-Polygon(v::AbstractVector{Tuple{Int, Int}}) = Polygon([Point(p...) for p in v])
-Polygon(v::AbstractVector{CartesianIndex{2}}) = Polygon([Point(p) for p in v])
-Polygon(rectangle::RectanglePoints) = Polygon([rectangle.p1, Point(rectangle.p2.x, rectangle.p1.y), rectangle.p2, Point(rectangle.p1.x, rectangle.p2.y)])
+Polygon(v::AbstractVector{Tuple{Int, Int}}; fill::Bool = false) = Polygon([Point(p...) for p in v]; fill = fill)
+Polygon(v::AbstractVector{CartesianIndex{2}}; fill::Bool = false) = Polygon([Point(p) for p in v]; fill = fill)
+Polygon(rectangle::RectanglePoints; fill::Bool = false) = Polygon([rectangle.p1, Point(rectangle.p2.x, rectangle.p1.y), rectangle.p2, Point(rectangle.p1.x, rectangle.p2.y)]; fill = fill)
 
 function draw!(img::AbstractArray{T, 2}, polygon::Polygon, color::T) where T<:Colorant
     draw!(img, Path(polygon.vertices), color)
     draw!(img, LineSegment(first(polygon.vertices), last(polygon.vertices)), color)
+
+    if polygon.fill
+        polygonscanfill!(img, polygon, color)
+    end
+
+    img
+end
+
+function polygonscanfill!(img::AbstractArray{T, 2}, polygon::Polygon, color::T) where T <: Colorant
+    image_height, image_width = size(img)
```

Comment:
Could we use `axes` here and check both the minimum and maximum indices? Not all AbstractArrays start indexing at 1 (e.g., OffsetArrays.jl) The `clamp` arguments below would also have to change, of course.